test/plugin/test_out_secondary_file.rb

```diff
@@ -0,0 +1,393 @@
+        FileUtils.mkdir_p("#{TMP_DIR}/test_dir")
+        File.chmod(0777, "#{TMP_DIR}/test_dir")
+        create_driver %[directory #{TMP_DIR}/test_dir/foo/bar/]
+      end
+
+      assert_raise Fluent::ConfigError do
+        FileUtils.mkdir_p("#{TMP_DIR}/test_dir")
+        File.chmod(0555, "#{TMP_DIR}/test_dir")
+        create_driver %[directory #{TMP_DIR}/test_dir/foo/bar/]
+      end
+    end
+
+    test 'should be passed directory' do
+      assert_raise Fluent::ConfigError do
+        create_driver %[]
+      end
+
+      assert_nothing_raised do
+        create_driver %[directory #{TMP_DIR}/test_dir/foo/bar/]
+      end
+    end
+  end
+
+  def check_gzipped_result(path, expect)
+    # Zlib::GzipReader has a bug of concatenated file: https://bugs.ruby-lang.org/issues/9790
+    # Following code from https://www.ruby-forum.com/topic/971591#979520
+    result = ""
+    File.open(path, "rb") { |io|
+      loop do
+        gzr = Zlib::GzipReader.new(io)
+        result << gzr.read
+        unused = gzr.unused
+        gzr.finish
+        break if unused.nil?
+        io.pos -= unused.length
+      end
+    }
+
+    assert_equal expect, result
+  end
+
+  class DummyMemoryChunk < Fluent::Plugin::Buffer::MemoryChunk; end
+
+  def create_metadata(timekey=nil, tag=nil, variables=nil)
+    Fluent::Plugin::Buffer::Metadata.new(timekey, tag, variables)
+  end
+
+  def create_es_chunk(metadata, es)
+    DummyMemoryChunk.new(metadata).tap do |c|
+      c.concat(es.to_msgpack_stream, es.size) # to_msgpack_stream is standard_format
+      c.commit
+    end
+  end
+
+  sub_test_case 'write' do
+    setup do
+      @record = { key: "vlaue" }
+      @time = Time.parse("2011-01-02 13:14:15 UTC").to_i
+      @es = Fluent::OneEventStream.new(@time, @record)
+    end
+
+    test 'should output compressed file with gzip option' do
+      d = create_driver
+      c = create_es_chunk(create_metadata, @es)
+      path = d.instance.write(c)
+
+      assert_equal "#{TMP_DIR}/out_file_test.0.gz", path
+      check_gzipped_result(path, @es.to_msgpack_stream.force_encoding('ASCII-8BIT'))
+    end
+
+    test 'should be output plain text without gzip option' do
+      d = create_driver %[
+        directory #{TMP_DIR}/
+        basename out_file_test
+      ]
+      c = create_es_chunk(create_metadata, @es)
+      path = d.instance.write(c)
+
+      assert_equal "#{TMP_DIR}/out_file_test.0", path
+      assert_equal File.read(path), @es.to_msgpack_stream.force_encoding('ASCII-8BIT')
+    end
+
+    test 'path should be incremental without append option' do
+      d = create_driver
+      c = create_es_chunk(create_metadata, @es)
+      packed_value = @es.to_msgpack_stream.force_encoding('ASCII-8BIT')
+
+      5.times do |i|
+        path = d.instance.write(c)
+        assert_equal "#{TMP_DIR}/out_file_test.#{i}.gz", path
+        check_gzipped_result(path, packed_value)
+      end
+    end
+
+    test 'path should be the same with append option' do
```

Comment:
Do you mean "path should be unchanged if append options is enabled"?
Original subject is nonsense.